cpp/arcticdb/processing/operation_types.hpp

```diff
@@ -120,7 +120,7 @@ struct unary_arithmetic_promoted_type {
 };
 
 template <class LHS, class RHS, class Func>
-struct binary_arithmetic_promoted_type {
+struct type_arithmetic_promoted_type {
```

Comment:
Renaming since it is now used for more than just _binary_ arithmetic.

Reply:
IsIn/IsNotIn are still binary operations. The "arithmetic" part of the name just referred to the fact that it was covering addition/subtraction/multiplication/division. binary_promoted_type is probably more accurate

Reply:
IMO - they never were binary operations. The others (roughly) are mappings

f: N x N -> N

whereas IsIn maps,

f: N x [N...] -> N

which is not what binary operation typically means.

Reply:
I view it as one of the operands being a set, not a hill I'm willing to die on though